src/test/java/ru/hogwarts/school/service/StudentServiceTest.java

```diff
@@ -1,29 +1,66 @@
 package ru.hogwarts.school.service;
 
-import org.junit.jupiter.api.Assertions;
-import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
 import ru.hogwarts.school.model.Student;
+import ru.hogwarts.school.repositories.StudentRepository;
+
+import java.util.NoSuchElementException;
+import java.util.Optional;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
 
+@ExtendWith(MockitoExtension.class)
 class StudentServiceTest {
-    private final StudentService studentService = new StudentService();
+@Mock
+private  StudentRepository RepositoryMock;
+@InjectMocks
+private  StudentService out;
 
-    @BeforeEach
+/*    @BeforeEach
     public void setUp() {
         studentService.createStudent(new Student(1l, "Виталий", 25));
         studentService.createStudent(new Student(2l, "Елена", 30));
         studentService.createStudent(new Student(3l, "Дмитрий", 22));
         studentService.createStudent(new Student(4l, "Екатерина", 26));
         studentService.createStudent(new Student(5l, "Сергей", 28));
         studentService.createStudent(new Student(6l, "Дмитрий", 29));
+    }*/
+
+    @Test
+    public void createStudentTest() {
+        Student student = new  Student(1l,"Антон",30);
+        Mockito.when(RepositoryMock.save(student)).thenReturn(student);
+        assertThat(out.createStudent(student)).isEqualTo(student);
+    }
+    @Test
+    public void getStudentPositiveTest(){
+        Student student = new  Student(1l,"Антон",30);
+        Mockito.when(RepositoryMock.findById(1l)).thenReturn(Optional.of(student));
+        assertThat(out.getStudent(1l)).isEqualTo(student);
+    }
+    @Test
+    public void updateStudentPositiveTest(){
+        Student student = new  Student(1l,"Антон",30);
+        Mockito.when(RepositoryMock.save(student)).thenReturn(student);
+
+        Student expected = new  Student(1l,"Дмитрий",33);
+        assertThat(out.updateStudent(new  Student(1l,"Дмитрий",33))).isEqualTo(expected);
     }
```

Comment:
Ты же ожидаешь, что если передан Дмитрий, то и вернётся Дмитрий. Правильнее тогда будет так:
```suggestion
    @Test
    public void updateStudentPositiveTest(){
        Student expected = new  Student(1l,"Дмитрий",33);
        Mockito.when(RepositoryMock.save(expected)).thenReturn(expected);
        assertThat(out.updateStudent(new  Student(1l,"Дмитрий",33))).isEqualTo(expected);
    }
```

Reply:
Спасибо, по мокито стало понятно что не так делал.